File: src/main/java/uk/co/agware/carpet/exception/MagicCarpetException.java
package uk.co.agware.carpet.exception;

/**
 * Created by Philip Ward <Philip.Ward@agware.com> on 7/05/2016.
 */
public class MagicCarpetException extends Exception {

    public MagicCarpetException() {
    }

    public MagicCarpetException(String message) {
        super(message);
    }

    public MagicCarpetException(String message, Throwable cause) {
        super(message, cause);
    }

    public MagicCarpetException(Throwable cause) {
        super(cause);
    }

    public MagicCarpetException(String message, Throwable cause, boolean enableSuppression, boolean writableStackTrace) {
        super(message, cause, enableSuppression, writableStackTrace);
    }
}

File: src/main/kotlin/uk/co/agware/carpet/exception/MagicCarpetException.kt
package uk.co.agware.carpet.exception;

abstract class MagicCarpetException(message: String) : RuntimeException(message)

class MagicCarpetDatabaseException(message: String): MagicCarpetException(message)

class MagicCarpetParseException(message: String): MagicCarpetException(message)
